featureDevelopmentGotchas.md

```diff
@@ -0,0 +1,25 @@
+# Edge Case Spidey Sense Tips and Tricks - Things to keep in mind for code development/reviews:
+(aka this is the kind of shit that has bitten us in the past)
+
+## Future proof?
+- Are you building in convention instead of design?
+  - How easy would it be for an unfamiliar dev to use your code improperly?
+
+## Complete logic
+- Are there unaccounted for logical branches?
+  - Wherever there is an if, consider the else
+  - Use exhaustive switch cases, always.
+  - DO NOT PUT IN EMPTY CASES OR SWALLOW ERRORS - log at a minimum
+  - Kotlin specific - If we do the work inside of a ?.let{} then what happens if the object IS null? Did you think about that?!
+
+## Persistent Data
+- Do the changes affect the schema of any persistent data? A migration is necessary.
+- Do the changes affect the state of any persistent data? DB values, shared prefs, files, the external settings in the device? Consider the possible states that an existing system could already have/be in. Is a migration necessary?
```

Comment:
Do you want to talk about deprecations here at all? One of the things we've struggled with between platforms is deprecating something and both platforms knowing and responding to said deprecation.